server/src/main/java/org/elasticsearch/cluster/routing/allocation/shards/ShardsAvailabilityHealthIndicatorService.java

```diff
@@ -910,11 +949,17 @@ public ShardAllocationStatus(Metadata clusterMetadata) {
         }
 
         void addPrimary(ShardRouting routing, ClusterState state, NodesShutdownMetadata shutdowns, boolean verbose) {
-            primaries.increment(routing, state, shutdowns, verbose);
+            primaries.increment(routing, state, shutdowns, verbose, TimeValue.timeValueSeconds(0));
```

Comment:
Wiring the setting through the static methods is class is a bit nasty. Decided to leave it out of `addPrimary` and just pass a dummy value.

Reply:
For this, let's use `TimeValue.MINUS_ONE` as the "special" value here